pytest/lib/branches.py

```diff
@@ -18,7 +18,7 @@
 
 def current_branch():
     return os.environ.get('BUILDKITE_BRANCH') or subprocess.check_output([
-        "git", "rev-parse", "--symbolic-full-name", "--abbrev-ref", "HEAD"
+        "git", "rev-parse", "HEAD"
```

Comment:
NB: this is a drive-by fix. Previous command would just return a `HEAD` for me in some cases, which would make this script fail to restore the checkout back to the original state.

Reply:
Well, yeah, but now without `--symbolic-full-name` this will always return sha of the commit so if someone is at a branch foo, after they run the test they’ll end up at detached head. You are getting `HEAD` when you are on detached head. This needs a different solution. I’ll prepare a commit.